Azure/sonic-buildimage
# Dell S6100 Port I2C changes to 201811 branch

**- Why I did it**
To resolve ISMT timeout issue, I2C changes has to be ported from master branch.

**- How I did it**
Ported changes from master.

**- How to verify it**
Verify bit bang reset logs in the syslog, check whether all the I2C components are coming up and verify show interfaces transceiver commands
[201811_i2c.txt](https://github.com/Azure/sonic-buildimage/files/5057589/201811_i2c.txt)

**- Which release branch to backport (provide reason below if selected)**




**- Description for the changelog**

Added I2C enumeration as a service in PMON.
Added Bit bang reset mechanisms.

Note: 
Depends on Azure/sonic-utilities#740
**DO NOT MERGE UNTIL ABOVE PR IS ported for 201811 branch**

**- A picture of a cute animal (not mandatory but encouraged)**
